### DISCUSSION TITLE
MSA plugin in R Shiny jbrowseR

### DISCUSSION THREAD

**Discussion by davemcg** [GMOD/jbrowse-components#3125] [Q&A]
Is it possible to use jbrowseR just with the MSA plugin? Trying to make a tiny(ish) MSA viewer in an R package.

I tried setting the config.json to

```
{
    "plugins": [
    {
      "name": "MsaView",
      "url": "https://unpkg.com/jbrowse-plugin-msaview/dist/jbrowse-plugin-msaview.umd.production.min.js"
    }
  ]
}
```

And this is my test R shiny code

```
library(shiny)
library(JBrowseR)

ui <- fluidPage(
  titlePanel("JSON JBrowseR Example"),
  JBrowseROutput("browserOutput")
)

server <- function(input, output, session) {
  # using JBrowseR helper function to parse the config
  config <- json_config("./config.json")

  output$browserOutput <- renderJBrowseR(
    JBrowseR("JsonView",
             config = config
    )
  )
}

shinyApp(ui, server)

```

But I get an error about an assembly not being given. From some reading around the jbrowseR view *has* to use a linear view (hence the error about a missing assembly? So I can't use the MSA plugin with jbrowseR?

<img width="918" alt="Screen Shot 2022-08-05 at 3 25 47 PM" src="https://user-images.githubusercontent.com/10225430/183147235-a926d436-ca36-474c-985d-e78ecd1a7488.png">

**Comment by cmdcolin** [comment]
good question! I am not sure if there is a quick way to do it as it since it is pretty much assuming a lot about doing the linear genome view but this is a great idea and it would be great to support other view types

**Comment by davemcg** [comment]
Thanks for following up, I was afraid this would be non-trivial!
